src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -2741,40 +2849,28 @@ public static class Extensions
 }
 ";
             // Extra errors
-            ParseAndValidate(test,
-    // (6,33): error CS1107: A parameter can only have one 'this' modifier
-    //     public static void Foo(this this t) {}
-    Diagnostic(ErrorCode.ERR_DupParamMod, "this").WithArguments("this"),
-    // (6,39): error CS1001: Identifier expected
-    //     public static void Foo(this this t) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, ")"),
-    // (7,37): error CS1001: Identifier expected
-    //     public static void Foo(this int this) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, "this"),
-    // (7,37): error CS1003: Syntax error, ',' expected
-    //     public static void Foo(this int this) {}
-    Diagnostic(ErrorCode.ERR_SyntaxError, "this").WithArguments(",", "this"),
-    // (7,41): error CS1031: Type expected
-    //     public static void Foo(this int this) {}
-    Diagnostic(ErrorCode.ERR_TypeExpected, ")"),
-    // (7,41): error CS1001: Identifier expected
-    //     public static void Foo(this int this) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, ")"),
-    // (9,34): error CS1001: Identifier expected
-    //     public static void Foo(this t) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, ")"),
-    // (10,32): error CS1001: Identifier expected
-    //     public static void Foo(int this) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, "this"),
-    // (10,32): error CS1003: Syntax error, ',' expected
-    //     public static void Foo(int this) {}
-    Diagnostic(ErrorCode.ERR_SyntaxError, "this").WithArguments(",", "this"),
-    // (10,36): error CS1031: Type expected
-    //     public static void Foo(int this) {}
-    Diagnostic(ErrorCode.ERR_TypeExpected, ")"),
-    // (10,36): error CS1001: Identifier expected
-    //     public static void Foo(int this) {}
-    Diagnostic(ErrorCode.ERR_IdentifierExpected, ")"));
+            CreateCompilationWithMscorlibAndSystemCore(test).GetDeclarationDiagnostics().Verify(
+                // (10,32): error CS1100: Method 'Foo' has a parameter modifier 'this' which is not on the first parameter
+                //     public static void Foo(int this) {}
+                Diagnostic(ErrorCode.ERR_BadThisParam, "this").WithArguments("Foo").WithLocation(10, 32),
+                // (7,37): error CS1100: Method 'Foo' has a parameter modifier 'this' which is not on the first parameter
```

Comment:
>// (7,37): error CS1100: Method 'Foo' has a parameter modifier 'this' which is not on the first parameter [](start = 16, length = 105)

Consider asserting parse errors as well to make sure that error experience for the line 7 isn't degraded.